RecoParticleFlow/PFClusterProducer/plugins/PFClusterFromHGCalMultiCluster.cc

```diff
@@ -0,0 +1,67 @@
+#include "PFClusterFromHGCalMultiCluster.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "FWCore/Framework/interface/Event.h"
+
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
+
+void PFClusterFromHGCalMultiCluster::updateEvent(const edm::Event& ev) {
+  ev.getByToken(_clusterToken, _clusterH);
+}
+
+void PFClusterFromHGCalMultiCluster::buildClusters(
+    const edm::Handle<reco::PFRecHitCollection>& input,
+    const std::vector<bool>& rechitMask, const std::vector<bool>& seedable,
+    reco::PFClusterCollection& output) {
+  // USE HGCal Multiclusters instead
+
+  const auto& hgcalMultiClusters = *_clusterH;
+  auto const& hits = *input;
+
+  // for quick indexing back to hit energy
+  std::unordered_map<uint32_t, size_t> detIdToIndex(hits.size());
+  for (uint32_t i = 0; i < hits.size(); ++i) {
+    detIdToIndex[hits[i].detId()] = i;
+  }
+
+  for (const auto& mcl : hgcalMultiClusters) {
+    edm::Ref<std::vector<reco::PFRecHit> > seed;
```

Comment:
only seed detid is used below. Is it really necessary to move around the full ref?